src/transformStateWithClones.js

```diff
@@ -6,8 +6,32 @@
  *
  * @return {Object[]}
  */
+function deepClone(obj) {
+  return JSON.parse(JSON.stringify(obj));
+}
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = deepClone(state);
+
+  actions.forEach((action) => {
+    if (action.type === 'addProperties') {
+      // eslint-disable-next-line object-curly-newline
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => {
+        if (currentState.hasOwnProperty(key)) {
+          delete currentState[key];
+        }
+      });
+    } else if (action.type === 'clear') {
+      currentState = {};
+    }
+
+    result.push(deepClone(currentState));
+  });
```

Comment:
please use switch statement, when the cases are finite - [checklist](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) point 3